routes/api/product-routes.js

```diff
@@ -0,0 +1,157 @@
+const { Router } = require("express");
+
+const { Product, Category, Tag, ProductTag } = require("../../models");
+
+// The `/api/products` endpoint
+
+const router = Router();
+
+// get all products
+router.get("/", async (req, res) => {
+  // find all products
+  // be sure to include its associated Category and Tag data
+  try {
+    const data = await Product.findAll({
+      include: [
+        {
+          model: Category,
+          model: Tag,
+        },
+      ],
+    });
+
+    return res.json({ success: true, data });
+  } catch (error) {
+    logError("GET Category", error.message);
+    return res
+      .status(500)
+      .json({ success: false, error: "Failed to send response" });
+  }
+});
+
+// get one product
+router.get("/:id", async (req, res) => {
+  // find a single product by its `id`
+  // be sure to include its associated Category and Tag data
+  try {
+    const data = await Product.findByPk(req.params.id, {
+      include: [
+        {
+          model: Tag,
+          through: { model: ProductTag, attributes: [] },
+        },
+        {
+          model: Category,
+        },
+      ],
+    });
+
+    if (data) {
+      return res.json({ success: true, data });
+    }
+
+    return res
+      .status(404)
+      .json({ success: false, error: "Product does not exist" });
+  } catch (error) {
+    logError("GET Category", error.message);
+    return res
+      .status(500)
+      .json({ success: false, error: "Failed to send response" });
+  }
+});
+
+// create new product
+router.post("/", async (req, res) => {
+  /* req.body should look like this...
+    {
+      product_name: "Basketball",
+      price: 200.00,
+      stock: 3,
+      tagIds: [1, 2, 3, 4]
+    }
+  */
+  const product = await Product.create(req.body);
```

Comment:
You do not need to create a product twice here. This one can be deleted as you never use the `product` variable in your code.